internal/controllers/terraformlayer/controller.go

```diff
@@ -70,11 +70,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	locked, err := lock.IsLocked(ctx, r.Client, layer)
 	if err != nil {
 		log.Errorf("failed to get Lease Resource: %s", err)
-		return ctrl.Result{RequeueAfter: r.Config.Controller.Timers.OnError}, err
+		return ctrl.Result{RequeueAfter: layer.GetTimeBeforeRetry(r.Config.Controller.Timers.OnError)}, err
```

Comment:
This is not linked to the failure annotation and should be a static time for a retry, this could be a transient error, the exponential backoff should only be applied to terraform failures